integration/copy_test.go

```diff
@@ -137,14 +137,19 @@ func (s *CopySuite) TestCopySimple(c *check.C) {
 	out := combinedOutputOfCommand(c, "diff", "-urN", dir1, dir2)
 	c.Assert(out, check.Equals, "")
 
-	// docker v2s2 -> OCI image layout
+	// docker v2s2 -> OCI image layout with image name
 	// ociDest will be created by oci: if it doesn't exist
 	// so don't create it here to exercise auto-creation
-	ociDest := "busybox-latest"
+	ociDest := "busybox-latest-image"
+	ociImgName := "busybox"
 	defer os.RemoveAll(ociDest)
-	assertSkopeoSucceeds(c, "", "copy", "docker://busybox:latest", "oci:"+ociDest)
+	assertSkopeoSucceeds(c, "", "copy", "docker://busybox:latest", "oci:"+ociDest+":"+ociImgName)
 	_, err = os.Stat(ociDest)
 	c.Assert(err, check.IsNil)
+
+	// docker v2s2 -> OCI image layout without image name
+	ociDest = "busybox-latest-noimage"
+	assertSkopeoFails(c, ".*Error writing manifest: cannot save image with empty image.ref.name.*", "copy", "docker://busybox:latest", "oci:"+ociDest)
```

Comment:
Testing this scenario might detect problems earlier if it were a unit test in the `oci` transport; testing that here doesn’t hurt, sure; but shouldn’t the `busybox-latest-noimage` directory also be cleaned up? The error is reported in `PutManifest`, which means `PutBlob` has already written files into the directory AFAICS.

(Even the old code should have used `ioutil.TempDir` instead of writing to the current directory, but that’s of course independent of this PR and not a blocker for it.)

Reply:
Fixed.